common/compositor/factory.cpp

```diff
@@ -17,46 +17,106 @@
 #include "factory.h"
 #include "platformdefines.h"
 
-#ifdef USE_GL
 #include "glrenderer.h"
 #include "glsurface.h"
 #include "nativeglresource.h"
-#elif USE_VK
```

Comment:
Even though some hardware platforms cannot support Vulkan, we **can** still build/deploy a software stack that contains Vulkan bits, right? Detecting if support is available already happens at runtime during the call to vk_is_supported().

I  think there is definitely some value in reducing the number of nonstandard build configurations, and we seem to have an opportunity here.